src/m365/spo/commands/page/page-header-set.ts

```diff
@@ -160,24 +162,32 @@ class SpoPageHeaderSetCommand extends SpoCommand {
 
     request
       .get<{ IsPageCheckedOutToCurrentUser: boolean, Title: string; }>(requestOptions)
-      .then((res: { IsPageCheckedOutToCurrentUser: boolean, Title: string; }): Promise<void> => {
+      .then((res: { IsPageCheckedOutToCurrentUser: boolean, Title: string; }): Promise<ClientSidePageProperties | null> => {
         title = res.Title;
 
         if (res.IsPageCheckedOutToCurrentUser) {
-          return Promise.resolve();
-        }
+          const requestOptions: any = {
```

Comment:
If we want to avoid removing current page content, shouldn't we always request current page contents before updating the header rather than only if the page is checked out to the current user?

Reply:
The checkout API also returns you the page information. Same as what you get back when requesting the current page. That is why I request the current page information if the page is already checked out to you, if not, get check it out. Both result in the same information.

Reply:
Cool. Thanks for clarifying. I'll add a comment to the code to make this behavior clear.

Reply:
Great! Makes sense.